lib/src/modules/gates.dart

```diff
@@ -513,3 +513,64 @@ class Mux extends Module with InlineSystemVerilog {
     return '$control ? $d1 : $d0';
   }
 }
+
+/// A generic two-input bit index gate [Module].
+///
+/// It always takes two inputs and has one output of width 1.
+class IndexGate extends Module with InlineSystemVerilog {
+  late final String _a, _b, _y;
+
+  /// The primary input to this gate.
```

Comment:
Let's not expose `a` and `b` inputs outside of this gate since they shouldn't really be accessed outside of the module.  I know some other modules do this, we should remove those too eventually.